### PR TITLE
Add further content for Attach API

### PR DESCRIPTION
**Pull request by SueChaplain**
Create new topic for Attach API with informatiom about usage, security, configuration.
Modify system property topics for accuracy, completeness and remove links to older material elsewhere.

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
Jenkins doc stage

File: docs/attachapi.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+A number of system properties are available to configure the Attach API when you start a Java application, as shown in the following table:
+
+| System property                                                                             |    Description                                                           |
+|---------------------------------------------------------------------------------------------|--------------------------------------------------------------------------|
+| [`-Dcom.ibm.tools.attach.directory=<directory_name>`](comibmtoolsattachdirectory.md)        | Specify a different directory in which to create `.com_ibm_tools_attach` |
+| [`-Dcom.ibm.tools.attach.displayName=<my_display_name>`](dcomibmtoolsattachdisplayname.md)  | Change the display name recorded by an agent                             |
+| [`-Dcom.ibm.tools.attach.id=<my_vm_ID>`](dcomibmtoolsattachid.md)                           | Change the VM identifier recorded by an agent                            |
+| [`-Dcom.ibm.tools.attach.timeout=<value_in_milliseconds>`](dcomibmtoolsattachtimeout.md)    | Change the connection timeout                                            |
+
```
Comment (JasonFengJ9) [inline docs/attachapi.md:79]: A few other system properties:
```
com.ibm.tools.attach.logging - Turn on tracing of attach API events. Valid values are "yes" and "no".
com.ibm.tools.attach.log.name - Override the default path for the log files generated by the "logging" option.
com.ibm.tools.attach.shutdown_timeout - Time delay before we give up trying to terminate the wait loop.
com.ibm.tools.attach.command_timeout - The socket timeout connecting to target VM.
```

File: docs/attachapi.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- z/OS systems must also set POSIX permissions on files and cannot rely on RACF&reg; or system level security to protect applications.
+- To avoid z/OS console messages reporting security violations in `/tmp`, add a security exception or specify a different common subdirectory by setting the `com.ibm.tools.attach.directory` system property.
+
+## Configuring the Attach API
+
+A number of system properties are available to configure the Attach API when you start a Java application, as shown in the following table:
+
+| System property                                                                             |    Description                                                           |
+|---------------------------------------------------------------------------------------------|--------------------------------------------------------------------------|
+| [`-Dcom.ibm.tools.attach.directory=<directory_name>`](comibmtoolsattachdirectory.md)        | Specify a different directory in which to create `.com_ibm_tools_attach` |
```
Comment (JasonFengJ9) [inline docs/attachapi.md:75]: I know this probably comes from an old internal doc during our slack conversation.
`Specify a different common directory for Attach API working files.` is correct, the specified directory is going to be the Attach API common subdirectory, i.e., replace `(new File(systemTmpDir,".com_ibm_tools_attach")).getPath()`.
Please remove this modification.

File: docs/dcomibmtoolsattachdirectory.md [hunk 3]
```diff
@@ -24,7 +24,7 @@
 
 # -Dcom.ibm.tools.attach.directory
 
-Specify a different common directory for Attach API working files.
+Specify a different directory in which to create the Attach API common subdirectory.
```
Comment (JasonFengJ9) [inline docs/dcomibmtoolsattachdirectory.md:27]: Please remove this modification.

File: docs/dcomibmtoolsattachenable.md [hunk 4]
```diff
@@ -52,13 +52,13 @@ A useful reference for information about the Java&trade; Attach API can be found
 
 > The Attach API is an extension that provides a mechanism to attach to a Java virtual machine. A tool written in the Java Language, uses this API to attach to a target  virtual machine and load its tool agent into that virtual machine.
 
-For example, to late attach the IBM&reg; Health Center agent to a virtual machine (VM) that is already running.
+A usage example is to late attach the IBM&reg; Health Center agent to a virtual machine (VM) that is already running. If the VM is started with the Attach API enabled, the Health Center agent can attach and monitor the running processes.
```
Comment (JasonFengJ9) [inline docs/dcomibmtoolsattachenable.md:56]: > the Health Center agent can attach and monitor the running processes.

To clarify, if the Health Center agent is started at the same time as the Java application, the agent and the application run in the same JVM, in this case, no need to attach to it in this case.

File: docs/attachapi.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+target application with the Attach API enabled. The Health Center agent can then attach to the VM running the target application and start monitoring.
+
+The OpenJ9 implementation of the Attach API is equivalent to the reference implementation. However, you can only use the Attach API to connect to another OpenJ9 VM.
+
+## Enabling the Attach API
+
+When you run a Java application, VM support for the Attach API is enabled by default on all platforms except z/OS&reg;. For security reasons on  z/OS,
+processes that use the default z/OS OMVS segment cannot enable the Attach API.
+
+To enable or disable the feature, set the `-Dcom.ibm.tools.attach.enable=[yes|no]` system property on the command line.
```
[the remark this answers is not in the document]
Comment (SueChaplain) [reply docs/attachapi.md:15]: Changed to " ...set the `-Dcom.ibm.tools.attach.enable=[yes|no]` system property command line option."

File: docs/attachapi.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
+# Java Attach API
+
+By using the Attach API library, your application can connect to a running VM and load an agent into that VM to run tasks. The typical use case for this feature is to load an agent that can be used to monitor the application that's running in the target VM.
```
[the remark this answers is not in the document]
Comment (SueChaplain) [reply docs/attachapi.md:3]: Modified to "With the Attach API, your application ....."

File: docs/attachapi.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+For example, if you want to monitor a target application with the IBM Health Center tool, start the  
+target application with the Attach API enabled. The Health Center agent can then attach to the VM running the target application and start monitoring.
```
[the remark this answers is not in the document]
Comment (SueChaplain) [reply docs/attachapi.md:6]: Thanks for explaining, now says:

For example, if you wanted to start monitoring an application that is already running with the Attach API enabled, you could use a tool such as the IBM Health Center. In this case, a Health Center agent can start in its own VM and attach to the target VM where the application  is running to start recording and sending data to the Health Center client.

File: docs/attachapi.md (hunk 2 above)
Comment (SueChaplain) [reply docs/attachapi.md:75]: Changed back. Also applied to the system property topic.

File: docs/dcomibmtoolsattachdirectory.md (hunk 3 above)
Comment (SueChaplain) [reply docs/dcomibmtoolsattachdirectory.md:27]: Done

File: docs/dcomibmtoolsattachenable.md (hunk 4 above)
Comment (SueChaplain) [reply docs/dcomibmtoolsattachenable.md:56]: Removed the second sentence "If the VM is started with the Attach API enabled, the Health Center agent can attach and monitor the running processes."

File: docs/attachapi.md (hunk 1 above)
Comment (SueChaplain) [reply docs/attachapi.md:79]: I assumed timeouts are in milliseconds. Can you provide a use case for the last 2 please? Are these just ways to shorten the `com.ibm.tools.attach.timeout` value when debugging?

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
Jenkins doc stage

File: docs/attachapi.md (hunk 1 above)
Comment (JasonFengJ9) [reply docs/attachapi.md:75]: Yes, the timeouts are in milliseconds.
`com.ibm.tools.attach.timeout` is the timeout for initial connection to attach the target VM;
`com.ibm.tools.attach.command_timeout` is the timeout for sending a command to the target VM after initial attachment; 
`com.ibm.tools.attach.shutdown_timeout` is the timeout for the termination of the Attach API wait loop thread.
These timeouts are effective in different stages.

File: docs/attachapi.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply docs/attachapi.md]: I would suggest something like _... use the -Dcom.ibm.tools.attach.enable=[yes|no] command line option to set the system property_. Feel free to change the wording though.

File: docs/attachapi.md (hunk 1 above)
Comment (JasonFengJ9) [reply docs/attachapi.md:75]: The default value for `com.ibm.tools.attach.timeout` is `120000`;
The default value for `com.ibm.tools.attach.command_timeout` is `0`, i.e., no time out;
The default value for `com.ibm.tools.attach.timeout` is `10000`.

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Pull request merged**